build.gradle

```diff
@@ -0,0 +1,331 @@
+			include "**/instance.*"
+			include "**/*.instance"
+			include "**/*.html"
+			include "**/*.css"
+			include "**/*.gif"
+			include "**/*.js"
+			include "**/*.png"
+			include "**/*.txt"
+			include "**/*.weirdsuffix"
+		}
+	}
+}
+
+tasks.withType(JavaCompile) {
+	classpath = configurations.allDeps
+	options.compilerArgs = ["-source", "11", "-target", "11", "-g"]
+}
+
+task clientJar(type: Jar, dependsOn: compileJava) {
+	from sourceSets.main.output
+	include "us/kbase/workspace/*.class"
+	exclude "us/kbase/workspace/WorkspaceServer.class"
+	include "us/kbase/common/service/*.class"
+	exclude "us/kbase/common/service/JsonServer*"
+	archiveFileName = clientJarFile
+	destinationDirectory = file("$dist/client")
+}
+
+task serviceJar(type: Jar, dependsOn: compileJava) {
+	from sourceSets.main.output
+	archiveFileName = serviceJarFile
+	destinationDirectory = file("$dist")
+}
+
+task buildJars {
+	dependsOn serviceJar, clientJar
+}
+
+task unzipJar(type: Copy) {
+	from zipTree("$jarDir/kbase/common/$kbaseCommonJar")
+	into "$tempunpack"
+}
+
+// by default, the files go into build/docs/javadoc
+tasks.javadoc {
+	dependsOn unzipJar
+	classpath = configurations.allDeps
+	failOnError = false
+	options {
+		author = false
+		noDeprecated = false
+		noDeprecatedList = false
+		noIndex = false
+		noNavBar = false
+		noTree = false
+		splitIndex = true
+		use = true
+		version = true
+		links = [
+			"http://download.oracle.com/javase/11/docs/api/",
+			"https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-core/2.9.9/",
+			"https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/2.9.9/"
+		]
+	}
+	source = fileTree(dir: "$src/us/kbase",
+	include: [
+		"workspace/*.java",
+		"common/service/*.java"
+	],
+	exclude: [
+		"workspace/WorkspaceServer.java",
+		"common/service/JsonServer*",
+		"common/service/JacksonTupleModule.java",
+		"common/service/JsonClientCaller.java",
+		"common/service/JsonTreeTraversingParser.java",
+		"common/service/KBaseJsonParser.java"
+	])
+	source += fileTree(dir: "$tempunpack/us/kbase/common/service", include: [
+		"UObject.java",
+		"JsonTokenStream.java",
+		"*Exception.java"
+	])
+
+	doLast {
+		delete "$tempunpack"
+	}
+}
+
+tasks.war {
+	dependsOn javadoc
+	description = "Build the WAR file. Assumes compilation has been run."
+	archiveFileName = warFile
+	destinationDirectory = file("$dist")
+
+	// location of the web.xml file
+	webXml = file("$warXml")
+
+	duplicatesStrategy "exclude"
+
+	// add the dependencies to lib
+	into("WEB-INF/lib") {
+		from sourceSets.main.compileClasspath
+	}
+	// copy the compiled classes over
+	into("WEB-INF/classes") {
+		from sourceSets.main.output.classesDirs
+	}
+	// copy in the documentation
+	into("WEB-INF/classes/server_docs/javadoc") {
+		from javadoc.destinationDir
+	}
+}
+
+task expandWar(type: Copy) {
+	from zipTree("$dist/$warFile")
+	into "war_contents"
+}
+
+// test reports are saved to build/reports/tests/<testName>
+tasks.withType(Test) {
```

Comment:
## Running Tests

Currently there is a single test task that runs all tests except the `*LongTest` tests. To execute it, run `gradle testNoLongTests` (or `gradle coverageNoLongTests` to generate coverage files). HTML test results are saved to the directory `build/reports/tests/<testName>` and coverage data (in CSV, XML, and HTML formats) goes to `build/reports/jacoco/<taskName>`.

### to test:
download one of the [`coverageNoLongTests-3.6.n3` archives from GitHub Actions](https://github.com/kbase/workspace_deluxe/actions/runs/4243574727#artifacts) to see an example of the resulting build directory and the test results/coverage.

### NOTE
Gradle is currently set to fork for each test run as there are two tests that fail if that setting is NOT on. To be continued!

Reply:
Assuming I'm reading this correctly we'll def want a target that runs all the tests, long tests included

Reply:
I have it set up so that in GitHub Actions, we would run the long tests, the mongo tests, and the quick tests (all tests minus long tests and mongo tests) in parallel, and then combine the coverage data. Locally it would be possible to run any of those sets on their own or all tests.